person.rb

```diff
@@ -32,14 +34,20 @@ def correct_name
 
 class Person < Nameable
   attr_reader :id
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def add_rental(date, book)
+    book_rent = Rental.new(date, book, self)
+    @rentals.push(book_rent)
   end
 
```

Comment:
- [ ] Kindly remove this unnecessary **line 50** which is `@rentals.push(book_rent)` from **add_rental** method to make your code clean :blush: Second it will automatically push your rental into the rentals array so you don't need to do that here manually :+1: